primary/tests/epoch_change.rs

```diff
@@ -67,17 +72,30 @@ async fn test_simple_epoch_change() {
     }
 
     // Move to the next epochs.
+    let mut old_committee = committee_0;
     for epoch in 1..=3 {
         // Move to the next epoch.
         let new_committee = Committee {
             epoch,
-            ..committee_0.clone()
+            ..old_committee.clone()
         };
 
-        for tx in &tx_channels {
-            tx.send(ConsensusPrimaryMessage::Committee(new_committee.clone()))
-                .await
-                .unwrap();
+        // Notify the old committee to change epoch.
+        let addresses: Vec<_> = old_committee
+            .authorities
+            .values()
+            .map(|authority| authority.primary.worker_to_primary.clone())
+            .collect();
+        let message = WorkerPrimaryMessage::Reconfigure(ReconfigureNotification::NewCommittee(
+            new_committee.clone(),
+        ));
+        let mut _do_not_drop = Vec::new();
+        for address in addresses {
+            _do_not_drop.push(
+                WorkerToPrimaryNetwork::default()
+                    .send(address, &message)
+                    .await,
```

Comment:
You could type the vector `let mut _do_not_drop: Vec<CancelHandle<()>> = Vec::new()`, that would help, and I have an upcoming refactor that helps identifying the behavior by renaming `CancelHandle` -> `CancelOnDropHandle`.

Reply:
good point, fixed